### PR TITLE
Extend MenuItem syntax to include tag

### PR DESCRIPTION
**Pull request by guergana**
added optional tag to MenuItem

We need this to show to the user which datatypes are not allowed.

Bug: T268441

### CODE REVIEW

File: vue-components/src/components/MenuItem.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export type MenuItem = {
 	label: string;
 	description: string;
+	tag?: string; // for internal use to signal disallowed datatypes
```
Comment (Ladsgroup) [inline vue-components/src/components/MenuItem.ts:4]: The comment here is misleading and confusing. The tag is not for disallowed datatypes, it's for datatypes with limited support and more importantly, that's for QueryBuilder and I expect reader of this component to understand the comment without even knowing what QB is.

**Review by Ladsgroup — CHANGES_REQUESTED** [review]

File: vue-components/src/components/MenuItem.ts (hunk 1 above)
Comment (guergana) [reply vue-components/src/components/MenuItem.ts:4]: you are completely right.

Comment (guergana) [reply vue-components/src/components/MenuItem.ts]: I changed it

**Comment by micgro42** [comment]
No, this task intended to include also extending the OptionsMenu to make use of that field and add styling for it, etc.
It should probably build on top of #284 to prevent merge conflicts.

I admit that the description of the Phabricator task is not ideal. I will update it.

**Comment by micgro42** [comment]
The code looks good, but the browser tests don't seem to be random failures. Even though I'm not sure why this PR should cause those tests to fail.

**Comment by guergana** [comment]
> The code looks good, but the browser tests don't seem to be random failures. Even though I'm not sure why this PR should cause those tests to fail.

I also saw that. I am trying to figure out why they are failing still.

**Comment by guergana** [comment]
It was a css property that doesn't have full browser support. Please re-review :)

**Review by micgro42 — APPROVED** [review]

File: vue-components/src/components/OptionsMenu.vue [hunk 2]
```diff
@@ -228,6 +238,18 @@ $base: '.wikit-OptionsMenu';
...
+		&__tag {
+			font-family: $wikit-OptionsMenu-item-tag-font-family;
+			font-size: $wikit-OptionsMenu-item-tag-font-size;
+			font-weight: $wikit-OptionsMenu-item-tag-font-weight;
+			color: $wikit-OptionsMenu-item-tag-font-color;
+			line-height: $wikit-OptionsMenu-item-tag-line-height;
+			font-style: italic;
+			padding-inline-start: $wikit-OptionsMenu-item-tag-left-spacing;
+			display: inline;
+			float: inline-end;
```
Comment (itamargiv) [inline vue-components/src/components/OptionsMenu.vue:251]: I think we forgot to remove that.

File: vue-components/src/components/OptionsMenu.vue [hunk 3]
```diff
@@ -19,13 +19,18 @@
...
 			ref="menu-items"
 		>
-			<div
-				:class="{
-					'wikit-OptionsMenu__item__label': true,
-					'wikit-OptionsMenu__item__label--bold': boldLabels,
-				}"
-			>
-				{{ menuItem.label }}
+			<div class="wikit-OptionsMenu__item__label-wrapper">
```
Comment (itamargiv) [inline vue-components/src/components/OptionsMenu.vue:22]: Nicely done!

File: vue-components/src/components/OptionsMenu.vue [hunk 4]
```diff
@@ -228,6 +238,18 @@ $base: '.wikit-OptionsMenu';
...
 			line-height: $wikit-OptionsMenu-item-description-line-height;
 		}
+
+		&__tag {
+			font-family: $wikit-OptionsMenu-item-tag-font-family;
+			font-size: $wikit-OptionsMenu-item-tag-font-size;
+			font-weight: $wikit-OptionsMenu-item-tag-font-weight;
+			color: $wikit-OptionsMenu-item-tag-font-color;
+			line-height: $wikit-OptionsMenu-item-tag-line-height;
+			font-style: italic;
```
Comment (itamargiv) [inline vue-components/src/components/OptionsMenu.vue:248]: @SaiSan-WMDE should there be a token for this?

File: vue-components/tests/unit/components/OptionsMenu.spec.ts [hunk 5]
```diff
@@ -50,4 +50,17 @@ describe( 'OptionsMenu', () => {
 
 		expect( wrapper.find( '.wikit-OptionsMenu__no-results' ).exists() ).toBeFalsy();
 	} );
+
+	it( 'displays the menu item tag in a menuItem', () => {
+		const menuItems = [
+			{ label: 'potato', description: 'root vegetable', tag: 'i am a tag' },
+			{ label: 'duck', description: 'aquatic bird' },
```
Comment (itamargiv) [inline vue-components/tests/unit/components/OptionsMenu.spec.ts:57]: :duck:

**Review by itamargiv — CHANGES_REQUESTED** [review]
LGTM! Just one really tiny nitpick

File: vue-components/src/components/OptionsMenu.vue (hunk 2 above)
Comment (SaiSan-WMDE) [inline vue-components/src/components/OptionsMenu.vue:251]:
```suggestion
			white-space: nowrap;
```

Comment (SaiSan-WMDE) [inline vue-components/src/components/OptionsMenu.vue:251]:
```suggestion
			float: inline-end;
			white-space: nowrap;
```

Comment (guergana) [reply vue-components/src/components/OptionsMenu.vue]: good catch. removed! thanks.

**Review by SaiSan-WMDE — APPROVED** [review]

File: vue-components/src/components/OptionsMenu.vue (hunk 3 above)
Comment (guergana) [reply vue-components/src/components/OptionsMenu.vue:22]: :100:

**Review by itamargiv — APPROVED** [review]
Thank you!

Some guy shouting tag

**Review by Ladsgroup — APPROVED** [review]

**Pull request merged**